github/repos_contents.go

```diff
@@ -91,6 +93,38 @@ func (s *RepositoriesService) GetReadme(owner, repo string, opt *RepositoryConte
 	return readme, resp, err
 }
 
+// DownloadContents can be used to download files.  It circumvents the normal
+// GitHub API (which has a file size limit) and uses the `download_url` link
+// provided by the API.  That URL provides a direct link to the raw data
+// associated with that version of the file and is not subject to the same
+// limits as the GitHub API based requests.
+//
+// If path references a file or a symlink and the call succeeds, it
+// returns an instance of io.ReadCloser that provides the contents of
+// the file.  Otherwise, it returns an error.
+func (s *RepositoriesService) DownloadContents(owner, repo, filepath string, opt *RepositoryContentGetOptions) (io.ReadCloser, error) {
```

Comment:
Yes, that is what I did at first.  But if you do that, you shadow the `path` import which is required to extract the file name and parent directory.  So there are three options here:

   1. Change the argument to `path` and `import` the `path` package under a different name.

   2. Change all other occurrences to of `path` as arguments to something else so they are all consistent with each other.

   3. Change just the argument to this function.

I thought 1 would be confusing since it would effectively rename (in the scope of this function) a standard package to something else.  I chose 3 because, in fact, this really must be the path to a file (whereas in other cases it could be either a directory or a filename).

If you want me to use 1 or 2 or if you have another options, let me know.